### PR TITLE
Env variables: increase size limit

### PR DESCRIPTION
**Pull request by stsewd**
Closes https://github.com/readthedocs/readthedocs.org/issues/7901

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--11748.org.readthedocs.build/en/11748/

- Developer's documentation (`dev`): https://dev--11748.org.readthedocs.build/en/11748/

### CODE REVIEW

File: readthedocs/projects/validators.py [hunk 1]
```diff
@@ -227,3 +230,19 @@ def _clean_prefix(prefix):
...
+
+
+def validate_environment_variable_size(
+    project, new_env_value, error_class=ValidationError
+):
+    existing_size = sum(
+        project.environmentvariable_set.annotate(size=Length("value")).values_list(
+            "size", flat=True
+        )
+    )
```
Comment (humitos) [inline readthedocs/projects/validators.py:242]: We could do `.aggregate(Sum("size"))` here if we want to do everything at db level.

File: readthedocs/api/v3/serializers.py [hunk 2]
```diff
@@ -1131,6 +1131,25 @@ class Meta:
...
+        )
+        return super().create(validated_data)
+
+    def update(self, instance, validated_data):
+        validate_environment_variable_size(
+            project=instance.project,
+            new_env_value=validated_data["value"],
+            error_class=serializers.ValidationError,
+        )
+        return super().update(instance, validated_data)
```
Comment (humitos) [inline readthedocs/api/v3/serializers.py:1152]: Do we need these validations here? Won't this be validated automatically due that you already defined the `clean()` method in the model itself?

**Review by humitos — APPROVED** [review]
Looks good to me 👍🏼

File: readthedocs/api/v3/serializers.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/api/v3/serializers.py:1152]: No, DRF doesn't run the clean method.

**Pull request merged**